pkg/util/k8s-client.go

```diff
@@ -1,96 +1,168 @@
+     }
+     fmt.Fprintf(os.Stderr, "ERROR: could not get default project from kubeconfig: %v", err)
+     // see if we are running in pod, or the NAMESPACE env var is set
+     b, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+     currentProject = string(b)
+     if err != nil {
+          fmt.Fprintf(os.Stderr, "error fetching token from k8s pod: %s", err.Error())
+          currentProject = os.Getenv("NAMESPACE")
+     }
+     if len(currentProject) == 0 {
+          currentProject = os.Getenv("POD_NAMESPACE")
+          klog.Infof("sidecar pod namespace is %s", currentProject)
+     }
+     return currentProject
 }
 
 func GetCurrentToken(cfg *rest.Config) string {
-	token := cfg.BearerToken
-	if len(token) == 0 {
-		b, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
-		token = string(b)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "error fetching token from k8s pod: %s", err.Error())
-			token = os.Getenv("K8S_TOKEN")
-		}
-	}
-	return token
+     if cfg == nil || len(cfg.BearerToken) == 0 {
+          b, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+          token := string(b)
+          if err != nil {
+               fmt.Fprintf(os.Stderr, "error fetching token from k8s pod: %s", err.Error())
+               token = os.Getenv("K8S_TOKEN")
+          }
+          return token
+     }
+     return cfg.BearerToken
+}
+
+// InClusterConfigHackForRHDHSidecars - between the RHDH operator forcing the RHDH deployment to have the setting of
+// automountServiceAccountToken: false
+// and patching into the volumes array using the coarse grained deployment object proving extremely difficult
+// we have had to get *VERY* hacky wrt supplying the REST config to controller runtime.  The method patterns
+// k8s.io/client-go/rest/config.go#func InClusterConfig() (*rest.Config, error) but checks env vars we set
+// on our various containers for our secret contents; fortunately, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT
+// are still set in the container; now, the root CA setup still wants a file, but it was possible to pass the
+// ephemeral "dynamic-plugins-root" volume to our sidecar, and then take the ca.crt env setting we get from
+// the bridge secret and store in a file we can seed into the TLS client config for our K8S REST Config.
+func InClusterConfigHackForRHDHSidecars() (*rest.Config, error) {
+     token := os.Getenv("token")
+     host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+     if len(host) == 0 || len(port) == 0 || len(token) == 0 {
+          return nil, rest.ErrNotInCluster
+     }
+
+     tlsClientConfig := rest.TLSClientConfig{Insecure: true}
+     cacrt := os.Getenv("ca.crt")
+     if len(cacrt) > 0 {
+          fn := "/opt/app-root/src/dynamic-plugins-root/ca.crt"
+          file, err := os.Create(fn)
+          klog.Infof("create ca.crt file err: %#v", err)
+          if err == nil {
```

Comment:
nit: we could use early returns with an error here to avoid having too much indentation.

Reply:
yep good call updated with latest push